pandas/tests/indexing/test_loc.py

```diff
@@ -2700,3 +2700,13 @@ def test_loc_setitem(self, string_series):
         string_series.loc[d2] = 6
         assert string_series[d1] == 4
         assert string_series[d2] == 6
+
+    @pytest.mark.parametrize("dtype", ["object", "string"])
+    def test_loc_assign_dict_to_row(self, dtype):
+        # GH41044
+        df = DataFrame({"A": ["abc", "def"], "B": ["ghi", "jkl"]}, dtype=dtype)
+        df.loc[0, :] = {"A": "newA", "B": "newB"}
+
+        expected = DataFrame({"A": ["newA", "def"], "B": ["newB", "jkl"]})
```

Comment:
Please add the dtype here too, then you can remove the check_dtype=False flag

Reply:
I was wondering if that was desirable but will do